src/main/java/org/apache/commons/lang3/text/StrSubstitutor.java

```diff
@@ -753,6 +757,7 @@ public boolean replaceIn(final StrBuilder source) {
      * @param offset  the start offset within the array, must be valid
      * @param length  the length within the builder to be processed, must be valid
      * @return true if altered
+     * @throws StringIndexOutOfBoundsException if {@code offset < 0 || offset + length >= source.length}
```

Comment:
It does strike me as possible that some of these exceptions shouldn't be thrown at all. If offset or length are out of bounds, maybe nothing is replaced.